packages/user-service/src/helpers.ts

```diff
@@ -35,7 +38,34 @@ class UserGroupApiHelper {
       });
     });
   }
+
+  // Helper function for the database interaction with ldap
+  public addUserLDAP ( profile_param: string ) {
+    return new Promise( ( resolve, reject ) => {
+      this.getProfilesBy( profile_param ).then( ( response: any ) => {
+        if ( !_.isEmpty( response ) ) {
+          const groups: any = [];
+          response.memberOf.map( ( group: any ) => {
+            groups.push( group.substring( group.indexOf( `cn=` ) + 3, group.indexOf( `,` ) ) );
+          } );
+          const newUser = new User( {
+            uid: response.uid,
+            name: response.cn,
+            rhatUUID: response.rhatUUID,
+            title: response.title,
+            memberOf: groups,
+            isActive: true,
+            apiRole: ( groups.includes( 'one-portal-devel' ) ) ? `ADMIN` : 'USER',
+            createdBy: response.rhatUUID,
+            createdOn: moment.utc( new Date() )
+          } );
+          resolve( newUser.save() );
+        } else {
+          resolve( {} );
```

Comment:
If the response is empty, shouldn't it reject the Promise?
Instead of resolving with an empty object.

Reply:
Updated